sysstat.service.in

```diff
@@ -2,20 +2,27 @@
 # (C) 2012 Peter Schiffer (pschiffe <at> redhat.com)
 #
 # @PACKAGE_NAME@-@PACKAGE_VERSION@ systemd unit file:
-#	 Insert a dummy record in current daily data file.
-#	 This indicates that the counters have restarted from 0.
+#     1. Insert a dummy record in current daily data file when starting/stopping
+#        the unit. This indicates that the counters have restarted from 0.
+#     2. This unit is automatically stopped/started on suspend/resume,
+#        see /usr/lib/systemd/system-sleep/sysstat-sleep.sh.
+#     3. Starting this unit also starts the timer unit for periodic accounting.
 
 [Unit]
-Description=Resets System Activity Logs
+Description=System Activity Accounting
+# Start the periodic accounting process
+Requires=sysstat-collect.timer
+# but make sure we finish first, because the accounting process locks the datafile
+Before=sysstat-collect.timer sysstat-collect.service
 
 [Service]
 Type=oneshot
 RemainAfterExit=yes
 User=@CRON_OWNER@
-ExecStart=@SA_LIB_DIR@/sa1 --boot
+ExecStart=@SA_LIB_DIR@/sa1 --boot -C STARTED
```

Comment:
> you cannot be absolutely sure that sadc will be called before long after reboot, long enough for every monotonic counters to have exceeded the value the had before system restart.

You can always be sure that if the counter is **lower** than the previous value, then a reboot has happened.

If the counter is **higher** than the previous value, then yes, you can't be sure if a reboot has happened. But you are /almost/ positive it hasn't. Because If the service is properly enabled and started on every boot then sadc should be called no sooner than when it was called after the previous boot, in the vast majority of cases.

If I understand what you're saying is about the following corner case:

*  sa1 is set up to be called every 10min.
* 00:08 reboot happens, counters are reset
* 00:10 an entry is recorded, counter value: 2min*60=120s (I'm using seconds after boot as an example)
* 00:12 second reboot happens, counters are reset
* 00:20 an entry is recorder, counter value: 8min*60=480s - monotonic counter increased, i.e. no reboot is recorded.

Still the vast majority of reboots should be handled fine by the auto-detection logic for the monotonic counters. Is there such logic?